### ISSUE TITLE
NaN handling is incosistent

### ISSUE THREAD

**Issue by HagaiHargil** [napari/napari#771]
## 🐛 Bug

The way napari currently handles `np.nan` values in its data is by raising wrong exceptions in some cases, and other times it kinda still works but not really.

## To Reproduce

```python
# The following two examples error due to some
# downstream error that the nan values create
image = np.array([[0, 1], [np.nan, 0]])
with napari.gui_qt():
    v = napari.view_image(image)  # Raises an OverflowError

points = np.array([[1, 1], [np.nan, np.nan]])
with napari.gui_qt():
    v = napari.view_points(points)  # Another OverflowError

# This sample doesn't error, but keeps generating
# RuntimeWarning when doing certain actions
with napari.gui_qt():
    v = napari.view_image(image)
    v.add_points(points)
```

## Expected behavior

Matplotlib and Seaborn usually show a single RuntimeWarning when encountering NaNs in the data, but they do still show the data and allow interactivity. I believe this is the behavior we should strive for, but it will probably require a considerable refactoring for many I/O operations (lazy loading, etc.).

Another option is to "fail fast" when encountering a NaN. However, in situations where this NaN is hidden deep inside the rendered dataset, stopping the app due to a NaN is very undesirable.

## Environment

```
napari: 0.2.6+8.gc0556a6 
Platform: Linux-4.19.85-1-MANJARO-x86_64-with-arch-Manjaro-Linux 
Python: 3.7.3 | packaged by conda-forge | (default, Jul 1 2019, 21:52:21) [GCC 7.3.0] 
PySide2: 5.13.2 
Qt: 5.13.2 
VisPy: 0.6.2 
NumPy: 1.17.3 
SciPy: 1.3.2 
scikit-image: 0.16.2 
Dask: 2.7.0 

GL version: '3.0 Mesa 19.2.6'
MAX_TEXTURE_SIZE: 16384 
```

**Comment by sofroniewn** [comment]
There has been some recent discussion of this on vispy https://github.com/vispy/vispy/pull/1769#issuecomment-560270948. Maybe @jni can weigh in here

**Comment by andy-sweet** [comment]
@HagaiHargil : I think is kind of an older duplicate of #1310, and was likely fixed by #3701 and friends, as I could not reproduce it on `main` (5c10022337601f350ad64ce56eddf6664306e40e).

If you agree, please close this issue! If not please explain why. Thanks again for writing up the issue.
